plugins/woocommerce/src/Internal/Features/FeaturesController.php

```diff
@@ -1086,4 +1088,44 @@ private function handle_plugins_page_views_list( $views ): array {
 			'incompatible_with_feature' => $incompatible_link,
 		);
 	}
+
+	/**
+	 * Set the feature nonce to be sent from client side.
+	 *
+	 * @param array $settings Component settings.
+	 *
+	 * @return array
+	 */
+	public function set_change_feature_enable_nonce( $settings ) {
+		$settings['_feature_nonce'] = wp_create_nonce( 'change_feature_enable' );
+		return $settings;
+	}
+
+	/**
+	 * Changes the feature given it's id, a toggle value and nonce as a query param.
+	 *
+	 * `/wp-admin/post.php?product_block_editor=1&_feature_nonce=1234`, 1 for on
+	 * `/wp-admin/post.php?product_block_editor=0&_feature_nonce=1234`, 0 for off
+	 */
+	private function change_feature_enable_from_query_params(): void {
+		if ( ! isset( $_GET['_feature_nonce'] ) || ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_GET['_feature_nonce'] ) ), 'change_feature_enable' ) ) {
```

Comment:
What message and title we should put there?

Reply:
This is a generic one we use in some places:

```
wp_die( esc_html__( 'Action failed. Please refresh the page and retry.', 'woocommerce' ) );
```

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/38551/commits/083eeb51f06a76fa7188dd770e1d99f03607fbf1

since this hook is called for every request we cannot `wp_die` anytime the nonce is not sent, so I moved it after validating that the intention is to change the feature flag.